rbd/encryption_opt_luks.go

```diff
@@ -0,0 +1,51 @@
+//go:build !octopus && !pacific && !quincy && ceph_preview
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// /* force XSI-complaint strerror_r() */
+// #define _POSIX_C_SOURCE 200112L
+// #undef _GNU_SOURCE
+// #include <stdlib.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// EncryptionOptionsLUKS generic options for either LUKS v1 or v2. May only be
+// used for opening existing images - not valid for the EncryptionFormat call.
+type EncryptionOptionsLUKS struct {
+	Passphrase []byte
+}
+
+func (opts EncryptionOptionsLUKS) allocateEncryptionOptions() cEncryptionData {
+	var cOpts C.rbd_encryption_luks_format_options_t
```

Comment:
So, here we still allocate Go memory, but it will be referenced in the `cEncryptionData`. ~I guess, wherever that is used, we still have the same problem.~ Ok, I checked, indeed it is used in a "safe" way, that is `cEncryptionData` is not passed directly as a C argument. But still:  I think it would be safer to `malloc` that as well and add it to the free function, because - as it happened to you this time - , it might happen again that someone falsely assumes, it is completely C allocated.

However: We can add that in a follow-up (I can do it) if you want to merge it as is for the release.

Reply:
Yes, but that's the one of the "old" methods, so I would prefer not to do that in this PR, but a follow up sounds good.